lib/auth/accountrecovery.go

```diff
@@ -103,66 +89,61 @@ func (a *Server) StartAccountRecovery(ctx context.Context, req *proto.StartAccou
 	return token, nil
 }
 
-// verifyCodeWithRecoveryLock counts number of failed attempts at providing a valid recovery code.
-// After MaxAccountRecoveryAttempts, user is temporarily locked from further attempts at recovering and also
-// locked from logging in. Modeled after existing function WithUserLock.
-func (a *Server) verifyCodeWithRecoveryLock(ctx context.Context, username string, recoveryCode []byte) error {
-	user, err := a.Services.GetUser(ctx, username, false)
-	switch {
-	case trace.IsNotFound(err):
-		// If user is not found, still authenticate. It should always return an error.
-		// This prevents username oracles and timing attacks.
-		return a.verifyRecoveryCode(ctx, username, recoveryCode)
-	case err != nil:
+// verifyRecoveryCodeWithRecord validates the recovery code for the user and will unlock their account if
+// the code is valid.  If the code is invalid, a failed recovery attempt will be recorded.
+func (a *Server) verifyRecoveryCodeWithRecord(ctx context.Context, username string, recoveryCode []byte) (errResult error) {
```

Comment:
```suggestion
func (a *Server) verifyRecoveryCode(ctx context.Context, username string, recoveryCode []byte) (errResult error) {
```

Now that you don't have two methods that do the same thing, we can simplify the name here.

Reply:
Updated, thank you